civicrm.php

```diff
@@ -1245,10 +1380,45 @@ public function remove_wp_magic_quotes() {
     $this->wp_request = $_REQUEST;
 
     // reassign globals
-    $_GET     = stripslashes_deep($_GET);
-    $_POST    = stripslashes_deep($_POST);
-    $_COOKIE  = stripslashes_deep($_COOKIE);
-    $_REQUEST = stripslashes_deep($_REQUEST);
+    $_GET     = stripslashes_deep($this->civi_get);
+    $_POST    = stripslashes_deep($this->civi_post);
+    $_COOKIE  = stripslashes_deep($this->civi_cookie);
+    $_REQUEST = stripslashes_deep($this->civi_request);
+
+    // test for query var
+    $q = get_query_var( 'q' );
+    if (!empty($q)) {
+
+		$page = get_query_var( 'page' );
+		$reset = get_query_var( 'reset' );
+		$id = get_query_var( 'id' );
+		$html = get_query_var( 'html' );
+		$snippet = get_query_var( 'snippet' );
+
+		$action = get_query_var( 'action' );
+		$mode = get_query_var( 'mode' );
+		$cid = get_query_var( 'cid' );
+		$gid = get_query_var( 'gid' );
+		$sid = get_query_var( 'sid' );
+		$cs = get_query_var( 'cs' );
+		$force = get_query_var( 'force' );
+
+		$_REQUEST['q'] = $_GET['q'] = $q;
+		$_REQUEST['page'] = $_GET['page'] = 'CiviCRM';
+		if (!empty($reset)) { $_REQUEST['reset'] = $_GET['reset'] = $reset; }
+		if (!empty($id)) { $_REQUEST['id'] = $_GET['id'] = $id; }
+		if (!empty($html)) { $_REQUEST['html'] = $_GET['html'] = $html; }
+		if (!empty($snippet)) { $_REQUEST['snippet'] = $_GET['snippet'] = $snippet; }
+
+		if (!empty($action)) { $_REQUEST['action'] = $_GET['action'] = $action; }
+		if (!empty($mode)) { $_REQUEST['mode'] = $_GET['mode'] = $mode; }
+		if (!empty($cid)) { $_REQUEST['cid'] = $_GET['cid'] = $cid; }
+		if (!empty($gid)) { $_REQUEST['gid'] = $_GET['gid'] = $gid; }
+		if (!empty($sid)) { $_REQUEST['sid'] = $_GET['sid'] = $sid; }
+		if (!empty($cs)) { $_REQUEST['cs'] = $_GET['cs'] = $cs; }
+		if (!empty($force)) { $_REQUEST['force'] = $_GET['force'] = $force; }
+
+    }
 
```

Comment:
Syntax: should be 2 spaces, not 2 tabs?

Reply:
Gah. Editor defaults to tab-indent. Will rectify.